### PR TITLE
feat(perforce): Add unit tests and fix terraform validation workflow

### PR DESCRIPTION
**Pull request by hwkiem**
## Summary

This PR adds comprehensive unit tests for the Perforce module and fixes the Terraform validation workflow to properly handle test directories.

### 1. Fixed Terraform Validation Workflow
- **File:** `.github/workflows/terraform-validation.yml`
- Removed `.tftest.hcl` from change detection filter (tests have separate workflow)
- Updated directory discovery to only find directories with `.tf` files
- Removed `grep -v '/tests$'` exclusion pattern (no longer needed)

**Result:**
- ✅ Validates all directories with `.tf` files (modules, submodules, examples, test setup)
- ✅ Skips directories with only `.tftest.hcl` files (test file directories)
- ✅ Simple, maintainable solution without complex exclusions

### 2. Reorganized Tests
- Created `tests/unit/` directory for mock-based unit tests
- Created `tests/integration/` directory for integration tests
- Moved existing integration tests and setup module to new structure
- Updated all path references in integration tests

### 3. Added Unit Tests

**`tests/unit/01_conditional_creation.tftest.hcl`** (8 test scenarios)
- Tests conditional creation of P4 Server, P4 Auth, and P4 Code Review submodules
- Validates ECS cluster creation logic
- Tests various combinations and edge cases

**`tests/unit/02_shared_resources.tftest.hcl`** (6 test scenarios)
- Tests shared ECS cluster creation
- Validates Route53 private hosted zone
- Tests load balancer access logging
- Verifies shared resource logic

**Features:**
- Mock providers (no AWS credentials required)
- Fast execution (seconds)
- Comprehensive assertions
- Self-contained tests

### 4. Added Documentation

- `tests/README.md` - Overview of test structure and workflow
- `tests/unit/README.md` - Unit test documentation
- `tests/integration/README.md` - Integration test documentation

## Testing

All pre-commit hooks passed:
- ✅ Security checks
- ✅ Terraform validation
- ✅ Terraform fmt
- ✅ Terraform tflint
- ✅ Link checking
- ✅ Markdown linting

## Benefits

- 🚀 Fast unit tests provide immediate feedback on conditional logic
- 🔒 No AWS credentials needed for unit tests
- 📊 14 test scenarios covering all module combinations
- 📚 Comprehensive documentation for contributors
- 🔧 Simplified validation workflow (no complex exclusions)

## Next Steps

After merge:
- Validation workflow will run on all `.tf` files
- Unit tests can be run locally with `terraform test -filter=tests/unit/`
- Integration tests continue to work with existing SSM parameters

### CODE REVIEW

File: .github/workflows/terraform-validation.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: Terraform Validation
 on:
   pull_request:
     paths:
-      - 'modules/**'
-      - 'samples/**'
+      - "modules/**"
```
Comment (gabebatista) [inline .github/workflows/terraform-validation.yml:5]: Why change this to double-quotes but leave the rest of the doc as single-quote?

Comment (hwkiem) [reply .github/workflows/terraform-validation.yml:5]: Ask Claude. TBH, not sure. I don't think it actually matters. It is a little odd that it didn't shift to double quotes for filters on workflow_dispatch section.

**Review by danielw-aws — APPROVED** [review]
